### DISCUSSION TITLE
Apollo server v4 migration question; Custom Directives

### DISCUSSION THREAD

**Discussion by alexbjorlig** [apollographql/apollo-server#7098] [Q&A] [answered]
I'm migrating to v4, and have problems with custom directives.

They don't seem to work. We have implemented custom directives based on the pattern described [here](https://www.apollographql.com/docs/apollo-server/v2/schema/creating-directives/#declaring-schema-directives).

In the apollo server v4, we do like this:

```
let schema = makeExecutableSchema({
  typeDefs: [
     directives,
     enums,
     // etc...     
  ]
}); 

schema = directive1(schema);
schema = directive2(schema);
schema = directive3(schema);

export default new ApolloServer({
   schema,
   // other options.
})
```

Has this changed with v4?

We are on the following versions:

```
"@graphql-tools/schema": "^9.0.5",
"@graphql-tools/utils": "^8.13.0",
"graphql": "^16.6.0",
"@apollo/server": "^4.0.4"
```

Thanks 🙏

**Comment by alexbjorlig** [comment, marked as answer]
Turned out that everything was actually working, sorry for the trouble.

**Reply by glasser** [reply]
Glad to hear it!

**Comment by ankush9598** [comment]
It still helps thanks !
